### ISSUE TITLE
Cross Tab Issue/ Doubt

### ISSUE THREAD

**Issue by PrasannaVenkatesh08** [SupremeTechnopriest/react-idle-timer#241]
## Question

Ask your question here!
My Scenerio is I have to show a prompt when timeout is passed and after certain time, I have to trigger the auto logout function.
For Example:
timeout = 1000 * 60 * 5 (5 mins)
promptTimeout = 1000 * 60 * 2 (2 mins)
also I should enable crossTabs and emitOnAllTabs because User should not be auto logged out if the user is active in one tab and idle in other tabs.
 
I am trying to implement the above feature with v5  but everything is working fine If the user is in one tab but It is not working If the user uses mulutple tabs.

Code :
```js
const timeout = 1000 * 30;
const promptTimeout = 1000 * 10;

function UserIdleTimeout() {
  const [openWarningModal, setOpenWarningModal] = useState(false);
  const [timer, setTimer] = useState(0);

  const onPrompt = () => {
    setOpenWarningModal(true);
    setTimer(Math.ceil(promptTimeout / 1000));
  };

  const onIdle = () => {
    setOpenWarningModal(false);
    setTimer(0);
    logout();
  };

  const onActive = () => {
    setOpenWarningModal(false);
    setTimer(0);
  };

  const { getRemainingTime, isPrompted, reset } = useIdleTimer({
    timeout,
    promptTimeout,
    onPrompt,
    onIdle,
    onActive,
    crossTab: true,
    emitOnAllTabs: true,
  });

  const stayLoggedIn = () => {
    setOpenWarningModal(false);
    reset();
  };

  useEffect(() => {
    const interval = setInterval(() => {
      if (isPrompted()) {
        setTimer(Math.ceil(getRemainingTime() / 1000));
      }
    }, 1000);
    return () => {
      clearInterval(interval);
    };
  }, [getRemainingTime, isPrompted]);

  return (
    <div>
      <Modal
        closeButtonProps={{ style: { display: "none" } }}
        isOpen={openWarningModal}
        onClose={() => {}}
        title={<div style={{ display: "none" }}></div>}
      >
        <div style={{ margin: "auto", width: "fit-content", marginBottom: 10 }}>
          You will be logged out in {timer} seconds!
        </div>
        <div style={{ margin: "auto", width: "fit-content" }}>
          <Button id="stayLoggedInID" variant="primary" size="medium" onClick={stayLoggedIn}>
            Stay Logged In
          </Button>
        </div>
      </Modal>
    </div>
  );
}
``` 
 
 Issue: When user uses mulltiple tab and also the user is idle in all the tabs --> After the timeout, the prompt is appeared but after the promptTimeout, onIdle function is not triggered

**Comment by SupremeTechnopriest** [comment]
Confirmed.  I will look into this this evening.

**Comment by PrasannaVenkatesh08** [comment]
Thanks @SupremeTechnopriest for taking this up quickly. Added to above one, Sometimes the timers in different tabs aren't sync like showing different times so that it affects the consistency of auto logout function and stay logged-in function.  Could you tell me how to use **syncTimers** for this. I tried with 1000(1 sec) but nothing seems to work for me.

**Comment by SupremeTechnopriest** [comment]
Set it to something lower like 200. The number that you set is how often it syncs. So at 1000 you will get one second of drift.

**Comment by PrasannaVenkatesh08** [comment]
Could you share me some examples of version 4 that have prompt feature and cross tab feature?

**Comment by SupremeTechnopriest** [comment]
In version 4 the prompt has to be implemented on your own.  Its a new feature in v5.  I should have it resolved this weekend.  I might need to refactor the way cross tab works. I've got some time blocked out for it tomorrow.

Cross Tab works in v4, but its configured a little differently.  There's no `syncTimers` option either.  The docs and examples for v4 can be found on the v4 branch. https://github.com/SupremeTechnopriest/react-idle-timer/tree/v4

**Comment by SupremeTechnopriest** [comment]
Did a rework of `crossTab`. Its published as `5.0.0-rc.24`.  It will solve your issue.  You can see the relevant changes on the docs here https://idletimer.dev/docs/getting-started/new#simplified-cross-tab-api.
